context/TracerContext.tsx

```diff
@@ -69,6 +69,16 @@ export const SelectedTracerStore: React.FC<StoreProps> = ({ tracer, children }:
 
     const [tracerState, tracerDispatch] = useReducer(reducer, initialState);
 
+    // Reset approval state on account change
+    const resetApproval = async () => {
+        selectedTracer?.removeApproved(selectedTracer.address);
+        await selectedTracer?.checkApproved(account);
```

Comment:
checkApproved already sets the approved values. you should be able to just call selectedTracer?.checkApproved(account) and remove the other two function calls

Reply:
That works. Thanks!